roles/roles.go

```diff
@@ -0,0 +1,94 @@
+package roles
+
+import (
+	"slices"
+
+	"github.com/sourcegraph/sourcegraph-accounts-sdk-go/services"
+)
+
+// Role is always the full qualified role name.
+type Role string
+
+// ToRole returns a role string in the format of
+// "service::role".
+func ToRole(service services.Service, name string) Role {
+	return Role(string(service) + "::" + name)
+}
+
+// ToStrings converts a list of roles to a list of strings.
+func ToStrings(roles []Role) []string {
+	ss := make([]string, len(roles))
+	for i, role := range roles {
+		ss[i] = string(role)
+	}
+	return ss
+}
+
+// ToRoles converts a list of strings to a list of roles.
+func ToRoles(strings []string) []Role {
+	roles := make([]Role, len(strings))
+	for i, s := range strings {
+		roles[i] = Role(s)
+	}
+	return roles
+}
+
+var (
+	// services.Dotcom
+	dotcomRoles = []string{
+		RoleNameDotcomSiteAdmin,
+	}
+)
+
+const (
+	// Role names for services.Dotcom
+
+	// Dotcom site admin
+	RoleNameDotcomSiteAdmin = "site_admin"
```

Comment:
We should only ever export fully qualified names:

```suggestion
	RoleDotComSiteAdmin = ToRole(service.DotCom, "site_admin")
```

Reply:
What do you think of https://github.com/sourcegraph/sourcegraph-accounts-sdk-go/pull/57/commits/dc28f5314ed9de439978dfa93d7c64aa2fa45f3a ?

In this case it probably makes sense that `ToRole` isn't exported